### PR TITLE
Fix shade warnings

### PR DESCRIPTION
**Pull request by lazerion**
- This PR attempts to solve `maven-shade-plugin:3.2.4:shade` warnings in package phase.

### CODE REVIEW

File: hazelcast-sql/pom.xml [hunk 1]
```diff
@@ -121,13 +126,16 @@
                                     <excludes>
                                         <!-- Avoid automatic discovery of Calcite JDBC. -->
                                         <exclude>META-INF/services/java.sql.Driver</exclude>
+                                        <exclude>META-INF/MANIFEST.MF</exclude>
```
Comment (lazerion) [inline hazelcast-sql/pom.xml:129]: @devozerov can we apply this for all artifacts, when I apply it to all artifacts I cannot see `MANIFEST.MF` though.

Comment (devozerov) [reply hazelcast-sql/pom.xml:129]: Currently, there are still warnings about it MANIFEST.MF overlap when building the `sql` module. Another thing is that packages from the `sql` module are not included in the resulting MANIFEST.MF. I am not sure how it should work in the end, but I would try to do the following for both `fat-shaded-jar` and `fat-jar` goals:
1. Exclude `MANIFEST.MF` from all packages (`<artifact>*:*</artifact>`)
1. Include `MANIFEST.MF` for the hazelcast-sql artifact
1. Make sure that `sql` manifest is merged into the final manifest using `ManifestResourceTransformer` [1]. I do not know where exactly this transformer should be defined, unfortunately

This way we will not have warnings and at the same time the final MANIFEST.ML will include packages from the `sql` module.

But I would also ask someone with more experience with Maven on how it should work in the end.

[1] https://maven.apache.org/plugins/maven-shade-plugin/examples/resource-transformers.html#ManifestResourceTransformer

**Comment by devozerov** [comment]
I reviewed the patch. There are several things that caught my attention:
1. There are still warnings when building the `sql` module about a clash with the `MANIFEST.MF` file. Perhaps, it should be fixed similarly to the core module - by excluding `MANIFEST.MF` from all artifacts
2. The resulting `hazelcast-all.jar` doesn't have packages from the `sql` module. Not sure whether this is a problem or not, but looks suspicious
3. Before 4.1 we had `apache-v2-licence.txt` file in the root directory. This is not the case now. Is it ok?
4. The `META-INF/LICENCE` file in the final JAR is ... **Jackson license**, not ours
5. Other JARs that we ship separately (e.g. `hazelcast-spring`, and soon `hazelcast-sql`) **do not have license file at all**!
6. In the current state, `hazelcast-sql.jar` doesn't have `MANIFEST.MF`

I realize that points 3-6 are not part of the current PR, and probably should be fixed/analyzed separately. Nevertheless, they are kind of related to this PR, because also concerned with MANIFEST/LICENSE/NOTICE files.

@jerrinot @Holmistr please share your thoughts on that

**Comment by devozerov** [comment]
The 3-6 questions are related to https://github.com/hazelcast/hazelcast/issues/17065. Added the reference to the above comment to the issue.

**Comment by lazerion** [comment]
@devozerov Thanks for the guidance, I will work on item-1 on your comments as per our discussion.

File: hazelcast-all/pom.xml [hunk 2]
```diff
@@ -111,11 +111,19 @@
                                 </transformer>
                                 <transformer
                                         implementation="org.apache.maven.plugins.shade.resource.ServicesResourceTransformer"/>
+                                <transformer
+                                        implementation="org.apache.maven.plugins.shade.resource.ApacheNoticeResourceTransformer"/>
+                                <transformer
+                                        implementation="org.apache.maven.plugins.shade.resource.DontIncludeResourceTransformer">
+                                    <resource>com.hazelcast.org.codehaus.commons.compiler.properties</resource>
+                                    <resource>hazelcast-community-license.txt</resource>
```
Comment (lazerion) [inline hazelcast-all/pom.xml:119]: @jerrinot @Holmistr We do not include `hazelcast-community-license.txt` in `hazelcast-all.jar` at all. It is regarding below warning 
```
[WARNING] hazelcast-aws-3.3.jar, hazelcast-wm-4.0.jar define 1 overlapping resource: 
[WARNING]   - hazelcast-community-license.txt
```
I think we do not need to include it inside the jar. kindly advise.

**Comment by Holmistr** [comment]
I'm removing myself as a reviewer. I will just be involved in what the requirements concerning licenses really are on the #licenses_in_jars Slack channel.

File: hazelcast-sql/pom.xml (hunk 1 above)
Comment (lazerion) [reply hazelcast-sql/pom.xml:129]: @devozerov updated with targeted filters, I could not make it possible with `ManifestResourceTransformer` FYI. PTAL.

**Comment by hz-highfive** [comment]
The job `Hazelcast-pr-builder` of your PR failed ([log](
https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/2536/build_log.txt), [artifacts](https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/2536/Hazelcast-pr-builder-2536.zip)). 
Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>
--------------------------
-------TEST FAILURE-------
--------------------------
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   MigrationInvocationsSafetyTest.migrationCommit_shouldRollback_whenTargetCrashes:324->PartitionCorrectnessTestSupport.assertSizeAndDataEventually:221->PartitionCorrectnessTestSupport.assertSizeAndDataEventually:225->HazelcastTestSupport.assertTrueEventually:1355->HazelcastTestSupport.assertTrueEventually:1253->PartitionCorrectnessTestSupport.access$000:66->PartitionCorrectnessTestSupport.assertSizeAndData:237->PartitionCorrectnessTestSupport.assertPartitionAssignments:179->PartitionCorrectnessTestSupport.assertPartitionAssignments:216 expected:<-1669084290458575971> but was:<8917940183757797572>
[INFO] 
[ERROR] Tests run: 31978, Failures: 1, Errors: 0, Skipped: 978
[INFO] 

[ERROR] There are test failures.
</details>
</pre>

**Review by devozerov — APPROVED** [review]
LGTM. There are no more warnings and the structure of the JARs appears to be correct.

**Comment by mmedenjak** [comment]
run-lab-run

**Comment by devozerov** [comment]
Reverting my approval. I performed a thorough analysis of the changes once again. Everything looks good to me except for the exclusion of the `hazelcast-community-license.txt` in the `hazelcast-all` module. The side effect of this change is that now we do not have the `hazelcast-community-license.txt` file in the resulting `hazelcast-akk.jar`. We do not have a consensus on how the licenses should look like in the final artifacts, so I propose to adjust the change in the `pom.xml` of the `hazelcast-all` module to a safer form:
1. Remove the line that excludes `hazelcast-community-license.txt`
1. Add an explicit filter for the `hazelcast-community-license.txt` file to either `hazelcast-aws-3.3` or `hazelcast-wm-4.0-sources` artifact.
This way we will avoid shade warnings about the duplicate license file, and at the same time, the license will be included in the final artifact.

**Review by devozerov — APPROVED** [review]

**Pull request merged**